petastorm/tests/test_spark_dataset_converter.py

```diff
@@ -0,0 +1,211 @@
+        StructField("str_col", StringType(), False),
+        StructField("bin_col", BinaryType(), False),
+        StructField("byte_col", ByteType(), False),
+    ])
+    df = test_ctx.spark.createDataFrame(
+        [(True, 0.12, 432.1, 5, 5, 0, "hello",
+          bytearray(b"spark\x01\x02"), -128),
+         (False, 123.45, 0.987, 9, 908, 765, "petastorm",
+          bytearray(b"\x0012345"), 127)],
+        schema=schema).coalesce(1)
+    # If we use numPartition > 1, the order of the loaded dataset would
+    # be non-deterministic.
+    expected_df = df.collect()
+
+    converter = make_spark_converter(df)
+    with converter.make_tf_dataset() as dataset:
+        iterator = dataset.make_one_shot_iterator()
+        tensor = iterator.get_next()
+        with tf.Session() as sess:
+            ts = sess.run(tensor)
+            # TODO: we will improve the test once the batch_size argument
+            #  added.
+            # Now we only have one batch.
+        for i in range(converter.dataset_size):
+            for col in df.schema.names:
+                actual_ele = getattr(ts, col)[i]
+                expected_ele = expected_df[i][col]
+                if col == "str_col":
+                    actual_ele = actual_ele.decode()
+                if col == "bin_col":
+                    actual_ele = bytearray(actual_ele)
+                assert expected_ele == actual_ele
+
+        assert len(expected_df) == len(converter)
+
+    assert np.bool_ == ts.bool_col.dtype.type
+    assert np.float32 == ts.float_col.dtype.type
+    assert np.float64 == ts.double_col.dtype.type
+    assert np.int16 == ts.short_col.dtype.type
+    assert np.int32 == ts.int_col.dtype.type
+    assert np.int64 == ts.long_col.dtype.type
+    assert np.object_ == ts.str_col.dtype.type
+    assert np.object_ == ts.bin_col.dtype.type
+
+
+def test_delete(test_ctx):
+    df = test_ctx.spark.createDataFrame([(1, 2), (4, 5)], ["col1", "col2"])
+    # TODO add test for hdfs url
+    converter = make_spark_converter(df)
+    local_path = urlparse(converter.cache_dir_url).path
+    assert os.path.exists(local_path)
+    converter.delete()
+    assert not os.path.exists(local_path)
+
+
+def test_atexit(test_ctx):
+    lines = """
+    from petastorm.spark.spark_dataset_converter import make_spark_converter
+    from pyspark.sql import SparkSession
+    import os
+    spark = SparkSession.builder.getOrCreate()
+    spark.conf.set('petastorm.spark.converter.defaultCacheDirUrl', '{temp_url}')
+    df = spark.createDataFrame([(1, 2),(4, 5)], ["col1", "col2"])
+    converter = make_spark_converter(df)
+    f = open(os.path.join('{tempdir}', 'test_atexit.out'), "w")
+    f.write(converter.cache_dir_url)
+    f.close()
+    """.format(tempdir=test_ctx.tempdir, temp_url=test_ctx.temp_url)
+    code_str = "; ".join(
+        line.strip() for line in lines.strip().splitlines())
+    ret_code = subprocess.call(["python", "-c", code_str])
```

Comment:
`python` -> `sys.executable`, since interpreter entry point could be different from the default `python` in `$PATH`.